### PR TITLE
Woo: Add product category create

### PR DESCRIPTION
**Pull request by coderkevin**
This adds actions and data-layer code for creating product categories via the WooCommerce API

To Test: `npm test`

### CODE REVIEW

File: client/extensions/woocommerce/state/action-types.js [hunk 1]
```diff
@@ -36,6 +36,8 @@ export const WOOCOMMERCE_PRODUCT_ACTION_LIST_CREATE = 'WOOCOMMERCE_PRODUCT_ACTIO
 export const WOOCOMMERCE_PRODUCT_ATTRIBUTE_EDIT = 'WOOCOMMERCE_PRODUCT_ATTRIBUTE_EDIT';
 export const WOOCOMMERCE_PRODUCT_CATEGORIES_REQUEST = 'WOOCOMMERCE_PRODUCT_CATEGORIES_REQUEST';
 export const WOOCOMMERCE_PRODUCT_CATEGORIES_REQUEST_SUCCESS = 'WOOCOMMERCE_PRODUCT_CATEGORIES_REQUEST_SUCCESS';
+export const WOOCOMMERCE_PRODUCT_CATEGORY_UPDATED = 'WOOCOMMERCE_PRODUCT_CATEGORY_UPDATED';
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/action-types.js]: These should be flipped to keep things in alphabetical order.

File: client/extensions/woocommerce/state/data-layer/product-categories/test/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+					siteId: 123,
+					onSuccessAction: productCategoryUpdated( 123, null, successAction ),
+					onFailureAction: failureAction,
+					body: { name: 'Category 1', slug: 'category-1' },
+				} )
+			);
+		} );
+	} );
+
+	describe( '#handleProductUpdated', () => {
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/data-layer/product-categories/test/index.js]: `handleProductCategoryUpdated`, not `handleProductUpdated`

File: client/extensions/woocommerce/state/action-types.js (hunk 1 above)
Comment (coderkevin) [inline client/extensions/woocommerce/state/action-types.js]: Fixed, thanks!

File: client/extensions/woocommerce/state/data-layer/product-categories/test/index.js (hunk 2 above)
Comment (coderkevin) [inline client/extensions/woocommerce/state/data-layer/product-categories/test/index.js]: Fixed, thanks!

**Pull request merged**